modules/@apostrophecms/asset/index.js

```diff
@@ -11,17 +11,103 @@ module.exports = {
 
   options: {
     alias: 'asset',
+    // If true, create both modern JavaScript and ES5 (IE11) compatibility
+    // builds of the ui/src browser code in each module, and serve them
+    // to the appropriate browsers without overhead for modern browsers.
+    // This does not attempt to compile the admin UI (ui/apos) for ES5.
+    es5: false,
     // If this option is true and process.env.NODE_ENV is not `production`,
     // the browser will refresh when the Apostrophe application
     // restarts. A useful companion to `nodemon`.
-    refreshOnRestart: false
+    refreshOnRestart: false,
+    // Frontend builds. Changing these settings is not recommended
+    // except as part of development of the Apostrophe core. To enable
+    // the src-es5 build just set the `es5: true` option of this module.
+    builds: {
+      src: {
+        scenes: [ 'public', 'apos' ],
+        webpack: true,
+        outputs: [ 'css', 'js' ],
+        label: 'public-facing modern JavaScript and SASS',
+        // Load index.js and index.scss from each module
+        index: true,
+        // Load only in browsers that support ES6 modules
+        condition: 'module'
+      },
+      'src-es5': {
+        // An alternative build from the same sources for IE11
+        source: 'src',
+        webpack: true,
+        scenes: [ 'public', 'apos' ],
+        // The CSS from the src build is identical, do not duplicate it
+        outputs: [ 'js' ],
+        label: 'public-facing modern JavaScript and SASS (IE11 build)',
+        // Load index.js and index.scss from each module
+        index: true,
+        // The polyfills babel will be expecting
+        prologue: stripIndent`
+          import "core-js/stable";
+          import "regenerator-runtime/runtime";
+        `,
+        // Load only in browsers that do not support ES6 modules
+        condition: 'nomodule'
+      },
+      public: {
+        scenes: [ 'public', 'apos' ],
+        outputs: [ 'css', 'js' ],
+        label: 'raw CSS and JS',
+        // Just concatenates
+        webpack: false,
+        prologue: stripIndent`
+          (function() {
+            window.apos = window.apos || {};
+            var data = document.body && document.body.getAttribute('data-apos');
+            Object.assign(window.apos, JSON.parse(data || '{}'));
+            if (data) {
+              document.body.removeAttribute('data-apos');
+            }
+          })();
+        `
+      },
+      apos: {
+        scenes: [ 'apos' ],
+        outputs: [ 'js' ],
+        webpack: true,
+        label: 'Apostrophe admin UI',
+        // Only rebuilt on npm updates unless CORE_DEV is set in the environment
+        core: true,
+        icons: true,
```

Comment:
I'm imagining some future installed module to add to the UI. It might have apps and components but no tiptap extensions. That step would simply continue quietly once it saw no tiptaps, right?